src/Security/Authorization/NoAuthenticationCredentialsHandlingAuthorizationChecker.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace eLife\Journal\Security\Authorization;
+
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException;
+
+final class NoAuthenticationCredentialsHandlingAuthorizationChecker implements AuthorizationCheckerInterface
```

Comment:
Better name?

Reply:
`FallbackAuthorizationChecker` with a configurable exception class name?
`MissingCredentialsAuthorizationChecker` for brevity?
If we knew why the credentials are missing in that context, that would shine light on the name

Reply:
I understand the test failing: it's lazily using the service constructed in the kernel (rather than using dummy dependencies), and as there's no request there's no firewall matched, so no credentials (not even an anonymous user). The same thing must happen with error states.

`MissingCredentialsAuthorizationChecker` sounds ok. Since it's very specific no need to make it configurable.